### PR TITLE
[Storage] [STG93 GA] APIView Feedback

### PR DESCRIPTION
**Pull request by vincenttran-msft**
As title states... pending some revision but need to locally gen and pass back into APIView to check for `metadata` issues going away, then need to check sphinx for the `upn` docstring change. Once that looks good, need to take the good pattern and propagate to all `upn` docstrings.

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py [hunk 1]
```diff
@@ -292,7 +292,8 @@ def get_directory_properties(self, **kwargs):
         :keyword bool upn:
             Optional. Valid only when Hierarchical Namespace is
             enabled for the account. If "True", the user identity values returned
-            in the x-ms-owner, x-ms-group, and x-ms-acl response headers will be
+            in the :attr:~DirectoryProperties.owner, ~DirectoryProperties.group,
+            and DirectoryProperties.permission response headers will be
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py:296]: Look at this in generated docs

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py [hunk 2]
```diff
@@ -966,8 +970,8 @@ def create_directory(self, directory_name, **kwargs):
         :param str directory_name:
             The name of the directory.
         :keyword metadata:
-            Name-value pairs associated with the directory as metadata.
-        :type metadata: dict[str, str]
+            Name-value pairs associated with the share as metadata.
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py:973]: `share` -> `directory`

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py [hunk 3]
```diff
@@ -570,8 +570,8 @@ async def start_copy_from_url(self, source_url, **kwargs):
                 This parameter was introduced in API version '2019-07-07'.
 
         :keyword metadata:
-            Name-value pairs associated with the file as metadata.
-        :type metadata: dict[str, str]
+            Name-value pairs associated with the share as metadata.
+        :paramtype metadata: dict[str, str]
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py:574]: Same nit

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py (hunk 1 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py:296]: Trying this again with https://stackoverflow.com/questions/27170958/sphinx-documentation-how-to-reference-a-python-property

File: sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py [hunk 4]
```diff
@@ -93,8 +93,7 @@ def readall(self):
             return data.decode(self._encoding)
         return data
 
-    def readinto(self, stream):
-        # type: (IO) -> None
+    def readinto(self, stream: IO[bytes]) -> None:
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py:96]: Check with Jacob on this -- if true, let's add a changelog entry 😄

File: sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py [hunk 5]
```diff
@@ -93,8 +93,7 @@ def readall(self):
...
         """Download the query result to a stream.
 
         :param IO stream:
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py:99]: Probably also fix up this type-hint if we are taking this change

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py (hunk 1 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py]: image

Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py]: Include fully qualified name EXCLUDING attr (but we can try it)

File: sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py (hunk 5 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py:99]: Could leave this out so it'll come through in my typing PR 😄

File: sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py (hunk 4 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py:96]: Going to kick this can down to the typing overhaul PR

**Review by jalauzon-msft — APPROVED** [review]

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py (hunk 1 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py]: ALRIGHTY, renders like this: 
image

I think this is the best approximation we can get here 😄 
:class:`~azure.storage.filedatalake.DirectoryProperties`

**Pull request merged**